temporal-test-server/src/main/java/io/temporal/internal/testservice/SelfAdvancingTimerImpl.java

```diff
@@ -207,27 +206,39 @@ public TimerLockHandle(LockEvent event) {
 
     @Override
     public void unlock() {
-      unlock(null);
+      lock.lock();
+      try {
+        unlockFromHandleLocked(event.caller + "[⎌]"); // caller releases its own lock
```

Comment:
The symbol is cute but hard to search for in logs. I'd just use some bit of text

Reply:
I used text before this symbol, this makes it easier to understand what's going on, especially when you look through longish lock histories. Mostly becuase lines can be logins sometimes to clearly identify the caller. Also, it makes it easier to search because it's a very specific symbol.